lib/Modules/FreeRdpHooks/FreeRdpHooks.psm1

```diff
@@ -24,33 +24,26 @@ Import-Module JujuHooks
 Import-Module OpenStackCommon
 
 function Get-Vcredist {
-    BEGIN {
+    Write-JujuWarning "Getting vcredist."
+    $vcredistUrl = Get-JujuCharmConfig -Scope "vcredist-url"
+    if(!$vcredistUrl) {
         $resourcesSupport = (Get-Command resource-get.exe -ErrorAction SilentlyContinue) -ne $null
```

Comment:
Indeed not the cleanest implementation. I agree with adding terminal exceptions to `Start-ExecuteWithRetry`